### PR TITLE
Refactor the Runtime Condenser to be less shit.

### PR DESCRIPTION
**Pull request by MrsTonedOne**
Runtime Condenser now supports src.loc.
Runtime Condenser now supports timestamped runtime logs (and supports older logs as well).
Runtime Condenser now supports unlimited runtimes. (good bye 100 runtime limit)
Runtime Condenser now uses expanding arrays so that it doesn't have to allocate a fuck ton of memory that may never get used.
Runtime Condenser now uses hash tables to make finding unique runtimes much faster. (n log m vs n^m).
Runtime Condenser now uses substr rather then find() where feasible to make that faster. 
Runtime Condenser now uses a non o(n^n) sorting algo that also required double memory (seriously, who wrote this? who the fuck is "Nodrak"?).
Runtime Condenser now uses a proper struct rather than SEVEN parallel arrays..
Runtime Condenser's compiled version should now work on machines without the C++ redistributable as well as "older" versions of windows like windows 7 =P.
Runtime Condenser now works on linux (and most likely mac) (the casing of Input.txt was wrong) just compile with `g++ -O3 -std=c++11 Main.cpp -o rc`
Runtime Condenser should be much faster as it was compiled with optimizations and automatic function inlining.

Removed the python version because it doesn't process a lot of non-standard output (and even some standard output) and its not needed as this version compiles on linux and even mac just fine.

### CODE REVIEW

File: tools/Runtime Condenser/Main.cpp [hunk 1]
```diff
@@ -155,261 +167,141 @@ bool readFromFile()
...
-		tempNumRuntime[i] = numRuntime[i];	numRuntime[i] = 0;
-	}
-
-	while(highestCount > 0)
-	{
-		for(int i=0; i <= maxStorage; i++) //For every runtime
-		{
-			if(tempNumRuntime[i] == highestCount) //If the number of occurances of that runtime is equal to our current highest
-			{
-				for(int j=0; j <= maxStorage; j++) //Find the next available slot and store the info
```
Comment (MrStonedOne) [inline tools/Runtime Condenser/Main.cpp]: ok, so let me break this down for you.

we take the highest count of runtime we've seen, decrease by 1, look thru every runtime, if it's count is the same, we loop thru every runtime, and find the next slot.

Thats O(m^n^n) where n is number of unique runtimes seen, and M is the highest count a unique runtime was seen.

File: tools/Runtime Condenser/Main.cpp [hunk 2]
```diff
@@ -13,139 +13,151 @@
...
-	//What we use to read input
-	string currentLine = "Blank";
-	string nextLine = "Blank";
-
-	//Stores lines we want to keep to print out
-	string storedRuntime[maxStorage+1];
-	string storedProc[maxStorage+1];
-	string storedSource[maxStorage+1];
-	string storedUsr[maxStorage+1];
-	string storedSrc[maxStorage+1];
```
Comment (MrStonedOne) [inline tools/Runtime Condenser/Main.cpp]: Thats not 1, not 2, not 3, but SIX parallel global arrays. (and one more further down)

File: tools/Runtime Condenser/Main.cpp [hunk 3]
```diff
@@ -155,261 +167,141 @@ bool readFromFile()
 				ss >> tmp >> c >> failures;
```
Comment (duncathan) [inline tools/Runtime Condenser/Main.cpp]: TODO: annihilate this awful awful line

File: tools/Runtime Condenser/Main.cpp (hunk 1 above)
Comment (MrPerson) [inline tools/Runtime Condenser/Main.cpp]: This is the kind of bullshit you expect to see when people share the worst code they've ever seen

Comment (optimumtact) [inline tools/Runtime Condenser/Main.cpp]: wtf

File: tools/Runtime Condenser/Main.cpp (hunk 3 above)
Comment (MrStonedOne) [inline tools/Runtime Condenser/Main.cpp]: I'll handle it for you.

File: tools/Runtime Condenser/Main.cpp (hunk 1 above)
Comment (duncathan) [inline tools/Runtime Condenser/Main.cpp]: https://tgstation13.org/parsed-logs/sybil/runtimes/runtime-2016-04-28-condensed.txt
this file would have taken 1407321^15646^^2 iterations
for a better sense of scale, 15646^^2 is a number with about 65 thousand digits

that's a lot of iterations

Comment (MrStonedOne) [inline tools/Runtime Condenser/Main.cpp]: Just 1407321^15646 is 96198 digits.

**Comment by Shadowlight213** [comment]
Noooo. Fuck. I couldn't ever get the non python version to run/compile on my machine.

**Pull request merged**